test_scripts/API/VehicleData/HandsOffSteering/common.lua

```diff
@@ -0,0 +1,309 @@
+--! isRequestOnHMIExpected: true - in case VehicleInfo.Sub/UnsubscribeVehicleData request on HMI is expected,
+--! otherwise - false
+--! pResultCode: resultCode value for expectation on App
+--! @return: none
+--]]
+function m.processSubscriptionRPCsSuccess(pRpcName, pAppId, isRequestOnHMIExpected, pResultCode)
+  local result = { success = true, resultCode = "SUCCESS", handsOffSteering = handsOffSteeringResponseData }
+  if not pAppId then pAppId = 1 end
+  if isRequestOnHMIExpected == nil then isRequestOnHMIExpected = true end
+  if not pResultCode then pResultCode = result end
+  local cid = m.getMobileSession(pAppId):SendRPC(pRpcName, { handsOffSteering = true })
+  if isRequestOnHMIExpected then
+    m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRpcName, { handsOffSteering = true })
+    :Do(function(_,data)
+      m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS",
+        { handsOffSteering = handsOffSteeringResponseData })
+    end)
+  else
+    m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRpcName):Times(0)
+  end
+  m.getMobileSession(pAppId):ExpectResponse(cid, pResultCode)
+  if pResultCode == result then
+    m.getMobileSession(pAppId):ExpectNotification("OnHashChange")
+    :Do(function(_, data)
+      m.setHashId(data.payload.hashID, pAppId)
+    end)
+  end
+end
+
+--[[ @processRPCHMIInvalidResponse: emulate incorrect HMI response for Subscribe/Unsubscribe RPC
+--! @parameters:
+--! @pRpcName: RPC name
+--! @pInvalidData: data for invalid response from HMI
+--! @return: none
+--]]
+function m.processRPCHMIInvalidResponse(pRpcName, pInvalidData)
+  local cid = m.getMobileSession():SendRPC(pRpcName, { handsOffSteering = true })
+  m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRpcName, { handsOffSteering = true })
+  :Do(function(_, data)
+    m.getHMIConnection():SendResponse(data.id, data.method, { handsOffSteering = pInvalidData })
+  end)
+  m.getMobileSession():ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
+  m.getMobileSession():ExpectNotification("OnHashChange") :Times(0)
+end
+
+--[[ @processRPCUnsuccessRequest: function for processing of unsuccess vehicle data request
+--! @parameters:
+--! @pRpcName: RPC name
+--! @pHandsData: parameters for the request
+--! @pResultCode: resultCode value for expectation on App
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.processRPCUnsuccessRequest(pRpcName, pHandsData, pResultCode, pAppId)
+  if not pAppId then pAppId = 1 end
+  local cid = m.getMobileSession(pAppId):SendRPC(pRpcName, { handsOffSteering = pHandsData })
+  m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRpcName) :Times(0)
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = pResultCode })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange") :Times(0)
+end
+
+--[[ @ignitionOff: IGNITION_OFF sequence
+--! @parameters: none
+--! @return: none
+--]]
+function m.ignitionOff()
```

Comment:
Please update `ignitionOff` function to more simple like in `common` file for smoke scripts